CMakeLists.txt

```diff
@@ -10,11 +10,99 @@ cmake_minimum_required(VERSION 3.16)
 if(NOT CMAKE_BUILD_TYPE)
     set(CMAKE_BUILD_TYPE "Debug")
 endif()
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+
+option(QT6 "Build with Qt6" OFF)
```

Comment:
Ah, or is the problem that you haven't downloaded Qt yet, so it can't know what major version it is?

In that case this option is only relevant when VCPKG is enabled? so doesn't need to be set on the Linux runners and can this option be behind the if(VCPKG) statement or is that not possible?

Reply:
Could potentially even be called VCPKG_ENABLE_QT6 or something to make it obvious that this isn't choosing to use Qt 6 for all platforms, but telling VCPKG which Qt versions to build and use?

Reply:
> Ah, or is the problem that you haven't downloaded Qt yet, so it can't know what major version it is?

That plus picking which version of Qt to use when both Qt5 and Qt6 are installed. Does specifying QT_VERSION_MAJOR as an argument to CMake work for picking which version to use when both are installed and using version-agnostic CMake code?

Reply:
Good question, I use containers to isolate my environments as different projects need different versions of everything :-D But I believe others use things like `qtchooser` which uses the QT_SELECT env variable to pick ? Or you just change your PATH to point to the qmake file of the version of Qt you want.

Then reading https://doc.qt.io/qt-6/cmake-qt5-and-qt6-compatibility.html it suggests that the CMake option of `QT_DEFAULT_MAJOR_VERSION` is the way to do it when you have both and then the find_package() will set `QT_VERSION_MAJOR` to the corresponding value.

Reply:
> QT_DEFAULT_MAJOR_VERSION

Oh, thanks! I'll use that variable to determine which Qt version to tell vcpkg to use.

Reply:
If it isn't set, should we just default to Qt 6 i wonder ? Eg default to the "latest and greatest" :-)

Reply:
(note this should only default for vcpkg itself and not on Linux where we want to to scan with find_package)